oci/multi_region.go

```diff
@@ -140,39 +140,44 @@ func BuildCompartementRegionList(ctx context.Context, d *plugin.QueryData) []map
 	return defaultMatrix
 }
 
-func getInvalidRegions(regions []string) []string {
-	ociRegions := []string{
-		"ap-chiyoda-1",
-		"ap-chuncheon-1",
-		"ap-hyderabad-1",
-		"ap-melbourne-1",
-		"ap-mumbai-1",
-		"ap-osaka-1",
-		"ap-seoul-1",
-		"ap-sydney-1",
-		"ap-tokyo-1",
-		"ca-montreal-1",
-		"ca-toronto-1",
-		"eu-amsterdam-1",
-		"eu-frankfurt-1",
-		"eu-zurich-1",
-		"me-dubai-1",
-		"me-jeddah-1",
-		"sa-santiago-1",
-		"sa-saopaulo-1",
-		"sa-vinhedo-1",
-		"uk-cardiff-1",
-		"uk-gov-cardiff-1",
-		"uk-gov-london-1",
-		"uk-london-1",
-		"us-ashburn-1",
-		"us-gov-ashburn-1",
-		"us-gov-chicago-1",
-		"us-gov-phoenix-1",
-		"us-langley-1",
-		"us-luke-1",
-		"us-phoenix-1",
-		"us-sanjose-1",
+// List out the regions supported by Oracle Cloud
+func listOciAvailableRegions(ctx context.Context, d *plugin.QueryData) ([]string, error) {
+	logger := plugin.Logger(ctx)
+	logger.Debug("listOciAvailableRegions")
+
+	cacheKey := "OciRegionList"
+	if cachedData, ok := d.ConnectionManager.Cache.Get(cacheKey); ok {
+		return cachedData.([]string), nil
+	}
+
+	// Create Session
+	session, err := identityService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	var regionNames []string
+
+	regions, err := session.IdentityClient.ListRegions(ctx)
```

Comment:
What happens if the user doesn't have access to this API? Do we fallback to a hardcoded list (like we do in AWS plugin)?

Reply:
@cbruno10, As per the current logic change, if the user doesn't have access to this API, it will not fall back to the hardcoded list.

Reply:
@cbruno10 we do not require the hardcoded list here; users with default access can call ListRegions API.